start_notebook.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+IMAGE_NAME=$1
+JUPYTER_TYPE=$2
+NOTEBOOK_DIR=$3
+PARTITION=$4
+
+IMAGE_DIRS=("/project/lgrandi/xenonnt/singularity-images" "/project2/lgrandi/xenonnt/singularity-images" "/dali/lgrandi/xenonnt/singularity-images")
+
+CONTAINER=""
+# If we passed the full path to an image and it exists, use that
+if [ -e "${IMAGE_NAME}" ]; then
+  CONTAINER="${IMAGE_NAME}"
+else
+  # Loop over the list of image directories
+  for IMAGE_DIR in "${IMAGE_DIRS[@]}"; do
+    # Check if the image exists in the current directory
+    if [ -e "${IMAGE_DIR}/${IMAGE_NAME}" ]; then
+      CONTAINER="${IMAGE_DIR}/${IMAGE_NAME}"
+      break
+    fi
+  done
+fi
+# if no container found, throw an error
+if [ -z "${CONTAINER}" ]; then
+  echo "Error: Singularity image not found. Please provide a valid image name or path."
+  exit 1
+fi
+
+if [ "x${JUPYTER_TYPE}" = "x" ]; then
+  JUPYTER_TYPE='lab'
+fi
+
+echo "Using singularity image: ${CONTAINER}"
+
+PORT=$((15000 + (RANDOM %= 5000)))
+
+if [[ "$PARTITION" == "lgrandi" || "$PARTITION" == "build" || "$PARTITION" == "caslake" ]]; then
+  SINGULARITY_CACHEDIR=/scratch/midway3/$USER/singularity_cache
+  SSH_HOST="midway3.rcc.uchicago.edu"
+  BIND_OPTS=("--bind /project2" "--bind /scratch/midway3/$USER" "--bind /project/lgrandi")
```

Comment:
Here you missed "--bind /project/lgrandi/xenonnt/dali:/dali"

Reply:
Resolved. Thanks for spotting!